staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -148,9 +150,10 @@ func (s *SecureServingOptions) ApplyTo(c *server.Config) error {
 		return nil
 	}
 
-	if s.BindPort <= 0 {
+	if err := s.CreateListener(); err != nil {
```

Comment:
we shouldn't mutate the options here. Let's check whether the listener is set, and if not create one further down for `c.SecureServingInfo.Listener = ...`.